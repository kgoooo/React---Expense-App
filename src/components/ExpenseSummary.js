import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expensesTotal';

const ExpenseSummary = ({expenseCount, expensesTotal}) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
	const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')
	return (
		<div>
			<h1>Currently displaying { expenseCount } { expenseWord } totalling { formattedExpensesTotal }.</h1>
		</div>
	)
}

const mapStateToProps = (state) => {
	const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);

	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: expensesTotal(visibleExpenses)
	}
}

export default connect(mapStateToProps)(ExpenseSummary);