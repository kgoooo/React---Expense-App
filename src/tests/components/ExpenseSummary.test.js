import React from 'react';
import { shallow } from 'enzyme'
import ExpenseSummary from '../../components/ExpenseSummary'

test('should correctly render expensesSummarry with one expense', () => {
	const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={999}/>);
	expect(wrapper).toMatchSnapshot()
})

test('should correctly render expensesSummarry with multiple expenses', () => {
	const wrapper = shallow(<ExpenseSummary expenseCount={3} expensesTotal={234999} />);
	expect(wrapper).toMatchSnapshot()
})
