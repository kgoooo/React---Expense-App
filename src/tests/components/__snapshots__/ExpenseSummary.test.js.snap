// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expensesSummarry with multiple expenses 1`] = `
<div>
  <h1>
    Currently displaying 
    3
     
    expenses
     totalling 
    $2,349.99
    .
  </h1>
</div>
`;

exports[`should correctly render expensesSummarry with one expense 1`] = `
<div>
  <h1>
    Currently displaying 
    1
     
    expense
     totalling 
    $9.99
    .
  </h1>
</div>
`;
